File: jmolecules-axon/02-onion-cqrs-es-bank/src/main/java/org/jmolecules/example/axonframework/domain/event/transfer/MoneyTransferRequestedEvent.java
package org.jmolecules.example.axonframework.domain.event.transfer;

import org.axonframework.serialization.Revision;

@Revision("1")
public record MoneyTransferRequestedEvent(
  String moneyTransferId,
  String sourceAccountId,
  String targetAccountId,
  int amount
) {
  // empty
}

File: jmolecules-axon/03-onion-axon-cqrs-es-bank/src/main/kotlin/org/jmolecules/example/axonframework/bank/domain/moneytransfer/event/MoneyTransferRequestedEvent.kt
package org.jmolecules.example.axonframework.bank.domain.moneytransfer.event

import org.axonframework.serialization.Revision
import org.jmolecules.event.annotation.DomainEvent
import org.jmolecules.example.axonframework.bank.domain.bankaccount.type.AccountId
import org.jmolecules.example.axonframework.bank.domain.bankaccount.type.Amount
import org.jmolecules.example.axonframework.bank.domain.moneytransfer.type.MoneyTransferId

@Revision("1")
@DomainEvent(namespace = "axon.bank", name = "MoneyTransferRequestedEvent")
data class MoneyTransferRequestedEvent(
  val moneyTransferId: MoneyTransferId,
  val sourceAccountId: AccountId,
  val targetAccountId: AccountId,
  val amount: Amount
)
